### PR TITLE
TRITON-1986 sdc-migrate not syncing delegated datasets in SmartOS and LX zones

### PR DESCRIPTION
**Pull request by twhiteman**

### CODE REVIEW

File: lib/workflows/vm-migration/begin.js [hunk 1]
```diff
@@ -187,6 +187,9 @@ function createProvisionPayload(job, cb) {
 
 
     var vmPayload = job.vmPayload = clone(job.params.vm);
+    var brand = vmPayload.brand;
+
+    assert.string(brand, 'brand');
```
Comment (kusor) [inline lib/workflows/vm-migration/begin.js]: This is gonna be run by workflow, if this throws an exception, you'll get no error message for the job results. I'd say we shouldn't use assert here and change to the approach below where you're returning an explicit error message when there's more than one dataset.

File: lib/workflows/vm-migration/begin.js [hunk 2]
```diff
@@ -214,9 +217,24 @@ function createProvisionPayload(job, cb) {
     delete vmPayload.pid;
     delete vmPayload.tmpfs;
 
+    // Handle delegated datasets for SmartOS and LX zones.
+    if (Array.isArray(vmPayload.datasets) && vmPayload.datasets.length > 0) {
+        assert.ok(brand === 'joyent' || brand === 'joyent-minimal' ||
```
Comment (kusor) [inline lib/workflows/vm-migration/begin.js]: Ditto, I wouldn't assert here. Check the brand and return an error if it's not the expected value.

File: lib/workflows/vm-migration/begin.js (hunk 1 above)
Comment (twhiteman) [reply lib/workflows/vm-migration/begin.js]: Yep - good point - will change to a callback error.

File: lib/workflows/vm-migration/begin.js (hunk 2 above)
Comment (twhiteman) [reply lib/workflows/vm-migration/begin.js]: I will change to a callback error.

**Pull request merged**